common/src/main/java/io/netty/util/internal/StringUtil.java

```diff
@@ -429,6 +429,29 @@ public static boolean isNullOrEmpty(String s) {
     }
 
     /**
+     * Find the index of the first non-white space character in {@code s} starting at {@code offset}.
+     * @param s The string to search.
+     * @param offset The offset to start searching at.
+     * @return the index of the first non-white space character or &lt;{@code 0} if none was found.
+     */
+    public static int indexOfNonWhiteSpace(String s, int offset) {
+        findnonwhitespace:
+        for (; offset < s.length(); ++offset) {
+            char c = s.charAt(offset);
+            switch(c) {
```

Comment:
This behavior is some different from Character.isWhitespace(). May be it makes sense to clarify this in javadoc?